src/store/middlewares/savedAccounts.js

```diff
@@ -27,10 +29,43 @@ const savedAccountsMiddleware = (store) => {
     }
   });
 
+  const updateSavedAccounts = (peers, tx, savedAccounts) => {
+    const { accounts } = savedAccounts;
+    tx.forEach((transaction) => {
+      const sender = transaction ? transaction.senderId : null;
+      const recipient = transaction ? transaction.recipientId : null;
+
+      accounts.forEach((account, i) => {
+        const address = extractAddress(account.publicKey);
+        const isSameNetwork = account.address === peers.data.options.address
+          && peers.data.options.code === account.network;
+
+        if ((address === recipient || address === sender) && isSameNetwork) {
+          getAccount(peers.data, address).then((result) => {
+            if (result.balance !== account.balance) {
+              accounts[i].balance = result.balance;
+              store.dispatch({ data: {
+                accounts,
+                lastActive: getLastActiveAccount(),
+              },
+              type: actionTypes.accountsRetrieved,
+              });
+            }
+          });
+        }
+      });
+    });
+  };
+
   return next => (action) => {
     next(action);
     const { peers, account, savedAccounts } = store.getState();
     switch (action.type) {
+      case actionTypes.newBlockCreated:
+        if (action.data.windowIsFocused) {
```

Comment:
If the window is focused afterward, there is again the problem of outdated balances. I would remove this condition.